tests/green_mode/filename_operationsTest.py

```diff
@@ -0,0 +1,111 @@
+        file_path = Path(__file__).parent / 'example_.project_data.yaml'
+        with file_path.open() as stream:
+            contents = yaml.load(stream)
+        files_list = get_files_list(contents['dir_structure'])
+        test_files_list = ['.coafile', 'example_file_1',
+                           'example_file_2', 'example_file_3',
+                           'example_file_4', 'example_file_5',
+                           'example_file_6']
+        self.assertEqual(files_list, test_files_list)
+
+    def test_check_filename_prefix_postfix(self):
+        # Files having a common prefix but no suffix.
+        file_path = Path(__file__).parent / 'example_.project_data.yaml'
+        with file_path.open() as stream:
+            contents = yaml.load(stream)
+        ret_val_contents = check_filename_prefix_postfix(deepcopy(contents),
+                                                         3, 3)
+        test_contents = deepcopy(append_to_contents(deepcopy(contents),
+                                                    'filename_prefix',
+                                                    ['example_file_']))
+        test_contents = append_to_contents(deepcopy(test_contents),
+                                           'filename_suffix',
+                                           [''])
+        self.assertEqual(test_contents, ret_val_contents)
+
+        # Works well when min files and min length of prefix are sufficiently
+        # low.
+        contents = {'dir_structure': ['py_some_name.xyz', 'py_some_other.c',
+                                      'py_yet_another_file.yaml']}
+        ret_val_contents = check_filename_prefix_postfix(deepcopy(contents),
+                                                         2, 2)
+        test_contents = deepcopy(append_to_contents(deepcopy(contents),
+                                                    'filename_prefix',
+                                                    ['py_']))
+        test_contents = append_to_contents(deepcopy(test_contents),
+                                           'filename_suffix',
+                                           [''])
+        self.assertEqual(test_contents, ret_val_contents)
+
+        # Works when min length of prefix and min files is exactly equal
+        # to the paramters passed.
+        ret_val_contents = check_filename_prefix_postfix(deepcopy(contents),
+                                                         3, 3)
+        test_contents = deepcopy(append_to_contents(deepcopy(contents),
+                                                    'filename_prefix',
+                                                    ['py_']))
+        test_contents = append_to_contents(deepcopy(test_contents),
+                                           'filename_suffix',
+                                           [''])
+        self.assertEqual(test_contents, ret_val_contents)
+
+        # Doesn't work when min length of prefix exceeds the prefix value.
```

Comment:
hmm, any reason why it fails for this case?

Reply:
I think this should be added as #FIXME

Reply:
See the the prefix that's gonna be detected from the files `['py_some_name.xyz', 'py_some_other.c', 'py_yet_another_file.yaml']` is `py_`, so when the minimum length of prefix is specified as 4, its not supposed to detect any prefix

Reply:
># Doesn't work when min length of prefix exceeds the prefix value.

So this means that when the *"allowed"* min length of prefix (4) exceeds the prefix value(3) it won't work/it won't detect it as a prefix